src/main/kotlin/uk/bfi/uvaudit/security/AuditUserService.kt

```diff
@@ -33,36 +35,52 @@ class AuditUserService(
         )
 
         val dbUserIds = jdbcTemplate.queryForList(sql, params, Long::class.java)
+        val userId = DataAccessUtils.uniqueResult(dbUserIds)
 
-        return when {
-            dbUserIds.isEmpty() -> {
-                null
-            }
-            dbUserIds.size == 1 -> {
-                AuditUser(dbUserIds[0], oidcUser)
-            }
-            else -> {
-                throw IncorrectResultSizeDataAccessException(0, dbUserIds.size)
-            }
+        return if (userId != null) {
```

Comment:
This can be `return userId?.let { AuditUser(it, oidUser) }`.

`let` is an extension function that takes `T` and maps it to `R` with a transform function, it's the Kotlin alternative to `Optional` when you have a nullable type -- it's a lot nicer since you can use it directly on `null` values.